WooCommerce/Classes/ViewRelated/Dashboard/Free Trial/StorePlanBannerPresenter.swift

```diff
@@ -57,32 +58,48 @@ final class FreeTrialBannerPresenter {
     /// Bring banner (if visible) to the front. Useful when some content has hidden it.
     ///
     func bringBannerToFront() {
-        guard let containerView, let freeTrialBanner else { return }
-        containerView.bringSubviewToFront(freeTrialBanner)
+        guard let containerView, let storePlanBanner else { return }
+        containerView.bringSubviewToFront(storePlanBanner)
     }
 }
 
-private extension FreeTrialBannerPresenter {
+private extension StorePlanBannerPresenter {
+    /// Observes the site info to add the expired banner as appropriate
+    ///
+    private func observeSiteInfo() {
+        ServiceLocator.stores.site
+            .sink { [weak self] site in
+                guard let self, let site else { return }
+                if !site.isWordPressComStore, site.wasEcommerceTrial {
+                    // Show plan expired banner for sites with expired WooExpress plans
+                    Task { @MainActor in
+                        await self.addBanner(contentText: Localization.expiredPlan)
```

Comment:
❓Should we also add code to remove the banner like in [the other method](https://github.com/woocommerce/woocommerce-ios/pull/10369/files#diff-faaf64c85f1bf164ad50a3b148503ab999290c7734d416d0799b5cb3fca4c553R99)?

Reply:
Good point! I decided to combine the site plan and site info checks to reuse the banner removal logic in 553cf2289c000ac06d23c9dd5fa38301371c5b24 and 94d86cf617e84eba0c12460ce90e7098eaba9a3f.